src/DatePickerInput.js

```diff
@@ -96,14 +96,8 @@ const DatePickerInput = React.createClass({
     }
   },
 
-  stopPropagation(e) {
-    if (this.props.closeOnClickOutside) {
-      e.stopPropagation();
-    }
-  },
-
-  hide() {
-    if (this.state.showing) {
+  hide(e) {
+    if (e.target !== this.refs.datePickerInput.getDOMNode() && this.state.showing) {
```

Comment:
`this.refs.x.getDOMNode()` is deprecated, please use `React.findDOMNode(this.refs.x)`